src/vibrato.rs

```diff
@@ -0,0 +1,153 @@
+use crate::ring_buffer::RingBuffer;
+use crate::lfo::LFO;
+
+pub struct Vibrato {
+    buffer: RingBuffer<f32>,
+    lfo: LFO,
+    delay: f32,
+    sample_rate: usize
+}
+
+impl Vibrato {
+    pub fn new(freq: f32, max_delay_secs: f32, sample_rate: usize) -> Self {
+        Vibrato {
+            buffer: RingBuffer::new((max_delay_secs * sample_rate as f32) as usize),
+            lfo: LFO::new(freq, sample_rate),
+            delay: 0.0,
+            sample_rate: sample_rate,
+        }
+    }
+
+    pub fn set_delay(&mut self, delay_in_secs: f32) {
+        if delay_in_secs < 0.0 {
+            panic!("Delay must be positive");
+        }
+        self.delay = delay_in_secs * self.sample_rate as f32;
+    }
+
+    pub fn set_freq(&mut self, freq: f32) {
+        if freq < 0.0 {
+            panic!("Vibrato freq must be positive");
+        }
+        self.lfo = LFO::new(freq, self.sample_rate);
+    }
+
+    pub fn process_block(&mut self, input: &[f32], output: &mut [f32]) {
```

Comment:
instantiate multiple vibrato instances for each channel. not sure if this is the best way. maybe pass in a slice of slice of each channel. and process all of them with one vibrato instance?